airflow/dag_processing/collection.py

```diff
@@ -393,19 +394,26 @@ def update_dags(
             dm.is_active = True
             dm.has_import_errors = False
             dm.last_parsed_time = utcnow()
-            dm.default_view = dag.default_view
             if hasattr(dag, "_dag_display_property_value"):
                 dm._dag_display_property_value = dag._dag_display_property_value
             elif dag.dag_display_name != dag.dag_id:
                 dm._dag_display_property_value = dag.dag_display_name
             dm.description = dag.description
-            dm.max_active_tasks = dag.max_active_tasks
-            dm.max_active_runs = dag.max_active_runs
-            dm.max_consecutive_failed_dag_runs = dag.max_consecutive_failed_dag_runs
-            dm.has_task_concurrency_limits = any(
-                t.max_active_tis_per_dag is not None or t.max_active_tis_per_dagrun is not None
-                for t in dag.tasks
-            )
+            # TODO: this `if is not None` is maybe not the best. It's convient though
+            if dag.max_active_tasks is not None:
+                dm.max_active_tasks = dag.max_active_tasks
```

Comment:
```
max_active_tasks = Column(Integer, nullable=False)
```

We shouldn't need the check right?

Reply:
I'll double check, it was something needed (I think) to cope with the lazy SerializedDag not being a full day, and so not having defaults fire things like this - that is, we don't encode defaults in SerializedDag json, so it's not there on this object and should take defaults.

This might be a bit hairy though